### PR TITLE
Enhance the System Fetcher with logic that create ORD webhook for some systems

### PR DESCRIPTION
**Pull request by kaloyanspiridonov**
**Description**

Changes proposed in this pull request:
- Enhance the System Fetcher with logic that create ORD webhook for some systems

**Related issue(s)**

- ...

**Pull Request status**

- [X] Implementation
- [X] Unit tests
- [X] Integration tests
- [X] `chart/compass/values.yaml` is updated 
- [X] Mocks are regenerated, using the automated script

### CODE REVIEW

File: components/director/internal/systemfetcher/service.go [hunk 1]
```diff
@@ -279,6 +282,21 @@ func (s *SystemFetcher) convertSystemToAppRegisterInput(ctx context.Context, sc
 		input.LocalTenantID = input.SystemNumber
 	}
 
+	if isOrdReady(sc.TemplateID) {
+		if input.BaseURL == nil || str.PtrStrToStr(input.BaseURL) == "" {
```
Comment (dragobt) [inline components/director/internal/systemfetcher/service.go:286]: add unit test for this case

**Review by dragobt — APPROVED** [review]

File: components/director/internal/domain/application/service.go [hunk 2]
```diff
@@ -1238,6 +1239,49 @@ func (s *service) genericUpsert(ctx context.Context, appTenant string, in model.
...
+	webhooks := make([]*model.Webhook, 0, len(appWebhooks))
+	for _, item := range appWebhooks {
+		webhooks = append(webhooks, item.ToWebhook(s.uidService.Generate(), appID, model.ApplicationWebhookReference))
+	}
+
+	webhooksToCreate := make([]*model.Webhook, 0, len(appWebhooks))
+	for _, wh := range webhooks {
+		found := false
+		for _, whDB := range webhooksFromDB {
+			if wh.Type == whDB.Type && wh.ObjectID == whDB.ObjectID && str.PtrStrToStr(wh.URL) == str.PtrStrToStr(whDB.URL) {
```
Comment (NickyMateev) [inline components/director/internal/domain/application/service.go:1268]: What's the point of `wh.ObjectID == whDB.ObjectID` being in the conditional?

Comment (NickyMateev) [reply components/director/internal/domain/application/service.go:1268]: Shouldn't the above `s.webhookRepo.ListByReferenceObjectID` guarantee that the webhooksFromDb will have this particular ObjectID?

Comment (kaloyanspiridonov) [reply components/director/internal/domain/application/service.go:1268]: honestly this check has been added for readability reasons.

**Review by dragobt — APPROVED** [review]

**Review by NickyMateev — APPROVED** [review]

**Pull request merged**